### PR TITLE
[WIP] Add support for symfony 5 #SymfonyHackday

### PR DESCRIPTION
**Pull request by vincentmoulene**
| Q             | A
| ------------- | ---
| Branch?       | "master" 
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

#SymfonyHackday
Pull request made for the support of Symfony 5. 
I am a beginner in open source, so let me know if everything is good.

I saw the tests failed with another component `symfony-cmf/testing` but I don't know how replace it in the `composer.json` file to test with another branch.
Let me know and I will update this PR.

Thanks

### CODE REVIEW

**Comment by dbu** [comment]
depends on https://github.com/symfony-cmf/Testing/pull/194

**Comment by dbu** [comment]
thank you for this pull request! we need to wait for the testing component to be updated to allow installation together with symfony 5.

**Comment by dbu** [comment]
i did some fixes for the build in master. could you please rebase on master (and solve the conflicts) to see if things now work?
it is quite likely that we need to do something about #238...

File: README.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ when building Symfony full stack applications.
 
 ## Requirements
 
-* PHP 7.1 / 7.2
-* Symfony 2.8 / 3.3 / 3.4 / 4.0
+* PHP 7.1 / 7.2 / 7.2
```
Comment (dbu) [inline README.md]: this should be 7.1 / 7.2 / 7.3 / 7.4

File: README.md [hunk 2]
```diff
@@ -31,8 +31,8 @@ when building Symfony full stack applications.
...
+* Symfony 3.3 / 3.4 / 4.0 / 5.0
```
Comment (dbu) [inline README.md]: for symfony 4, we only allow 4.3 / 4.4 - lets just delete the line and refer to composer.json, that will be easier to maintain.

File: composer.json [hunk 3]
```diff
@@ -16,16 +16,17 @@
...
         "symfony/phpunit-bridge": "^4.2.2",
-        "symfony/dependency-injection": "^3.4 || ^4.3",
-        "symfony/config": "^3.4 || ^4.3",
-        "symfony/event-dispatcher": "^3.4 || ^4.3",
-        "symfony-cmf/testing": "^3@dev"
+        "symfony/dependency-injection": "^3.4 || ^4.3 || ^5.0",
+        "symfony/config": "^3.4 || ^4.3 || ^5.0",
+        "symfony/event-dispatcher": "^3.4 || ^4.3 || ^5.0",
+        "symfony/browser-kit": "^3.4.22 || ^4.3 || ^5.0",
+        "symfony-cmf/testing": "^3.0.x-dev || dev-enhancement/symfony-5-support"
```
Comment (dbu) [inline composer.json]: you need the `@dev` or we would need to change to allow dev stability.

i think we can remove the `|| dev-enhancement/symfony-5-support`, right?

File: composer.json [hunk 4]
```diff
@@ -16,16 +16,17 @@
...
         "symfony/phpunit-bridge": "^4.2.2",
-        "symfony/dependency-injection": "^3.4 || ^4.3",
-        "symfony/config": "^3.4 || ^4.3",
-        "symfony/event-dispatcher": "^3.4 || ^4.3",
-        "symfony-cmf/testing": "^3@dev"
+        "symfony/dependency-injection": "^3.4.22 || ^4.1 || ^5.0",
+        "symfony/config": "^3.4.22 || ^4.1 || ^5.0",
+        "symfony/event-dispatcher": "^3.4.22 || ^4.1 || ^5.0",
+        "symfony/browser-kit": "^3.4.22 || ^4.1 || ^5.0",
+        "symfony-cmf/testing": "@dev"
```
Comment (dbu) [inline composer.json]: please keep this `^3@dev`

**Comment by acrobat** [comment]
@dbu I would like to help with the sf 5 support. But how would you like to solve the `UrlGeneratorInterface` parameter issue, without breaking BC and still support sf 4 and 5 in the current major?

**Comment by dbu** [comment]
hi @acrobat , thanks for the offer! yeah that parameter typing is indeed a major issue. afaik in the past we abused the weak type checking to do something that symfony had not intended.

i think the only way forward is to place the route object in the $attributes and use `""` as route name or something like that.

our router implementation will need some sort of version check to determine which method signature to use, with some PHP hackery like this: https://github.com/kuraobi/DoctrinePHPCRBundle/blob/acc7abfd5e3ea865ef77ef110f41232e28c58cc7/src/DependencyInjection/DoctrinePHPCRExtension.php#L572-L595

the old signature would take the route object from the $route parameter, but emit a deprecation warning, saying you should pass the route as attribute.

we need to adjust our code and templates to pass the route as attribute, as well as adjust the documentation.

would be great if you take a go at that. i pushed support-for-symfony-5 to this repository so that you can do pull requests against that branch.

**Comment by dbu** [comment]
finished in #250

**Pull request closed without merging**